src/widgets/Components/TimetableChanges.tsx

```diff
@@ -0,0 +1,193 @@
+import React, { forwardRef, useEffect, useImperativeHandle, useState, useCallback, useMemo } from "react";
+import { ActivityIndicator, Text, View } from "react-native";
+import { useTheme } from "@react-navigation/native";
+import { Clock, ClockAlert } from "lucide-react-native";
+
+import { WidgetProps } from "@/components/Home/Widget";
+import WidgetHeader from "@/components/Home/WidgetHeader";
+import ColorIndicator from "@/components/Lessons/ColorIndicator";
+import { getSubjectData } from "@/services/shared/Subject";
+import { TimetableClass, TimetableClassStatus } from "@/services/shared/Timetable";
+import { useCurrentAccount } from "@/stores/account";
+import { useTimetableStore } from "@/stores/timetable";
+import { dateToEpochWeekNumber } from "@/utils/epochWeekNumber";
+import { updateTimetableForWeekInCache } from "@/services/timetable";
+
+const lz = (num: number) => (num < 10 ? `0${num}` : num);
+
+const TimetableChanges = forwardRef(({ hidden, setHidden, loading, setLoading }: WidgetProps, ref) => {
+  const account = useCurrentAccount(store => store.account!);
+  const timetables = useTimetableStore(store => store.timetables);
+  const [changedCourse, setChangedCourse] = useState<TimetableClass | null>(null);
+
+  const currentWeekNumber = useMemo(() => dateToEpochWeekNumber(new Date()), []);
+
+  useImperativeHandle(ref, () => ({
+    handlePress: () => "Lessons"
+  }));
+
+  const fetchTimetable = useCallback(async () => {
+    if (!timetables[currentWeekNumber] && account.instance) {
+      setLoading(true);
+      try {
+        await updateTimetableForWeekInCache(account, currentWeekNumber);
```

Comment:
Force la mise à jour de l'emploi du temps pask actuellement, faut aller dans la page pour que ça mette à jour vers les dernières infos

```suggestion
        await updateTimetableForWeekInCache(account, currentWeekNumber, true);
```